lib/builders.js

```diff
@@ -79,18 +82,44 @@ class JSONBuilder {
 
   add(npmPackage) {
     const { name, version } = npmPackage;
-    const sources = npmPackage.licenseSources.license.sources;
 
-    const licenseText = formatLicenseText(sources.reduce((acc, source) => `${acc}${source.text}`, ''));
+    const licenseText = npmPackage.text ? formatLicenseText(npmPackage.text) : '';
 
     this.addCached(name, version, licenseText);
   }
 
+  buildLink(name, repository, licenseFile) {
+    return new Promise((resolve) => {
```

Comment:
Not sure what node version you're using but you can easily make this method `async` and get rid of the promise boiler plate

Reply:
Can do, just more used to promises (plus the original returns for `request` was a promise, so I was just sticking with that)